### DISCUSSION TITLE
Permissions needed for the Clickhouse

### DISCUSSION THREAD

**Discussion by Pranavk02** [rilldata/rill#4906] [Q&A] [answered]
I have made a user and given him read and list roles in clickhouse but unable to connect through Rill. But on the same side I have another user which have all the permissions of the clickhouse and with that I am able to query and make graphs so I wanted to know which specific permission should I give to my clickhouse user so that I can query tables and make dashboards

**Comment by k-anshul** [comment]
Hey @Pranavk02 

Could you check if the user can read data from `information_schema` database. You can try running below query:

```
SELECT
	T.table_schema AS SCHEMA,
	T.table_schema = currentDatabase() AS is_default_schema,
	T.table_name AS NAME,
	T.table_type AS TABLE_TYPE, 
	C.column_name AS COLUMNS,
	C.data_type AS COLUMN_TYPE,
	C.ordinal_position as ORDINAL_POSITION
FROM information_schema.tables T 
JOIN information_schema.columns C ON T.table_schema = C.table_schema AND T.table_name = C.table_name
WHERE lower(T.table_schema) NOT IN ('information_schema', 'system')
ORDER BY SCHEMA, NAME, TABLE_TYPE, ORDINAL_POSITION
```

**Reply by Pranavk02** [reply]
Hi @k-anshul 
Thanks for replying back!
Yes I had run the query and is able to read data.

**Comment by k-anshul** [comment, marked as answer]
I am able to create a dashboard with a user with following permissions (My tables are in default db).
```
CREATE USER readonly_user IDENTIFIED WITH plaintext_password BY 'your_password';
CREATE ROLE select_role;
GRANT SELECT ON default.* TO select_role;
GRANT select_role TO readonly_user;
```

**Comment by Pranavk02** [comment]
Thanks @k-anshul now I am able to connect! Thanks for the help😊
